### ISSUE TITLE
Dustwalker Priests Not Receiving #reanimpriest Properly

### ISSUE THREAD

**Issue by alguLoD** [elmokki/nationgen#940]
Looking at the files all undead priests are supposed to receive these, but Dustwalkers do not.

**Comment by Ealb** [comment]
Argh, I thought I fixed this. Like, I specifically went through undead
poses and made case-by-case should/shouldn't decisions (IIRC, everything
but ghouls I went with "should").

**Comment by alguLoD** [comment]
Oh, about ghouls: looks like advanced human ghoul nations don't generate properly at all, at least not filtering for "Advanced Humans" as the primary race and with the "ghouls" theme. IIRC they were in Dominions 4 so it's not an all-Colossus issue of being unimplemented.

**Comment by Ealb** [comment]
That's very odd, since I know I generated some randomly when testing the
rec point changes.

**Comment by alguLoD** [comment]
Might be me screwing something up, then. I'll check in an hour or so when I'm back home if you haven't already.

**Comment by elmokki** [comment]
If this is a real issue and the ghoul nation filter contains #primaryrace not foo or #secondaryrace not foo or something similar, that may be the case.

**Comment by Ealb** [comment]
Re: dustwalker priests, it looks like it's an engine issue. They've got #reanimpriest in the .dm...

**Comment by Ealb** [comment]
Waaaaaaaaait, that's a nation command, isn't it? Gaaaaah.

**Comment by elmokki** [comment]
Is it? Documentation is pretty bad.

**Comment by Ealb** [comment]
No, re-checking the modding manual it claims to be a unit command, albeit one still w/o documentation. In which case it's just not working.

**Comment by Ealb** [comment]
Had a chance to check Advanced Human ghould themes, BTW. They _seem_ fine, in that the files don't have anything obviously breakable and that I could set the filtering system to generate me 5 ghoul nations (even if it rejected slightly less than 600 to make 'em)

**Comment by alguLoD** [comment]
Right, I actually checked that again now, and it works fine with just the theme. What I was doing wrong was also filtering for "Advanced Human" as the primary race which does break it. I guess the theme does something to override that part.

**Comment by Ealb** [comment]
The interference from the race filter actually makes perfect sense, because I was trying to be too clever. There's a couple of demographic themes - ghouls from Adv. Humans, I think Cavernwights from Agarthans, maybe Lamia from Amazons - where I viewed it as conceptually a different race sharing the same definitions/themes/etc, so they overwrote the race field with a new one. I'm wavering on whether I should leave that as-is, get rid of the race reassignment, or just add a race file using the same items and theme files and get rid of the theme.

**Comment by elmokki** [comment]
While my principal thought here is that it really doesn't matter, it'd be best if races wouldn't be too heavily rewritten with themes, just for the sake of intuition.

**Comment by Ealb** [comment]
That was my thought as well. If we didn't have filtering during generation it would be fine, but since we do it's a problem.

**Comment by elmokki** [comment]
Though I suppose we could just make the filters use a separate file for race lists and such that contains ghoul separately. It'd be neat since right now the program initializes nationgen before GUI to get the data and that takes some time nowadays (though you could read the race files only if you wanted so arguably that, too, is a very crude solution)

**Comment by Ealb** [comment]
I'll just make them keep race the same, continue to change visible race, make sure subrace prefixes are defined properly (i.e. so a Human race that's all Ghouls doesn't try to call everything Ghoul Infantry - in retrospect this may have been what prompted me to do this originally), and stick a couple of theme tags into the nation. That should allow anything that would be facilitated by actually changing the race.

**Comment by Ealb** [comment]
I went ahead and changed ghouls and lamia to have their theme's containing race as their race plus a few relevant themes. I did direct-edit-and-commit since I'm away from home and don't have the GitHub client on my computer, so while it seems to have taken properly within the repository, the changes are not showing up looking at recent changes via pulse. FWIW.

**Comment by Legendsmith** [comment]
By the way, the ``#reanimpriest`` command works, it's just that it also needs a level, and an appropriate nation command such as ``#zombiereanim``

**Comment by Ealb** [comment]
That's a real mess. The list of "new undocumented Dom5 commands" lists it as taking no argument, and the number appears to be entirely irrelevant - a unit with reanimpriest 1 but H3 gets normal H3 reanimation numbers, while one with RA3 and H2 gets H2 numbers, and both RA1 + H1 and RA2 H1 give H1 nos, etc.

Also, no, despite the tooltip, you don't want #zombiereanim, etc. - that lets non-national priests reanimate. Just giving an argument is enough to get them to reanimate, though again the actually argument seems irrelevant...
